### PR TITLE
Custom logic to get URL from incoming ASP.NET Core request

### PR DESCRIPTION
**Pull request by yogiraj07**
*Issue #64 , if available:*

*Description of changes:*
As per recommendation from ASP.NET Core team on this [issue](https://github.com/aspnet/AspNetCore/issues/2718), adding custom logic to fetch URL from the incoming ASP.NET Core request. 

###Testing
Tested locally for ASP.NET Core 2.2 web app.
For no host present ` curl -v --http1.0 -H "Host:" http://localhost:5000/`, `http` section in the trace generated by the logic :

```json
"http": {
              "request": {
                        "url": "http:///",
                        "method": "GET",
                        "user_agent": "curl/7.55.0",
                        "client_ip": "::1"
                },
                "response": {
                        "status": 200
                }
            },
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: sdk/src/Handlers/AspNetCore/Internal/AWSXRayMiddleware.cs [hunk 1]
```diff
@@ -273,9 +275,32 @@ private static void PopulateRequestAttributes(HttpRequest request, Dictionary<st
             }
         }
 
-        private static string GetUrl(HttpRequest request)
+        // Implementing custom logic : https://github.com/aws/aws-xray-sdk-dotnet/issues/64
+        public static string GetUrl(HttpRequest request)
         {
-            return Microsoft.AspNetCore.Http.Extensions.UriHelper.GetDisplayUrl(request);
```
Comment (luluzhao) [inline sdk/src/Handlers/AspNetCore/Internal/AWSXRayMiddleware.cs]: From my understanding, in stead of calling `GetDisplayUrl`, you implementing a way to construct the url by itself? Why you don't want to use the method when the request is not NULL.

Comment (yogiraj07) [reply sdk/src/Handlers/AspNetCore/Internal/AWSXRayMiddleware.cs]: This is related to this [issue](https://github.com/aws/aws-xray-sdk-dotnet/issues/64) 

The NET Core 2.x throws null pointer exception on `GetDisplayUrl()` call, when `Host` header is not present in the incoming request. This is a bug from the downstream dependency which is fixed in NET Core 3.x (not yet GAed).

I talked to ASP NET Core community and they provided to copy paste the code : [link](https://github.com/aspnet/AspNetCore/issues/2718), see my comments in the last section. 

The request is not null is just a safety check, however the issue lies deeper inside the `request` object. Looking at the current situation, writing custom code gives fine grain control on null safety checks on individual parameters of the request instance.

Comment (luluzhao) [reply sdk/src/Handlers/AspNetCore/Internal/AWSXRayMiddleware.cs]: I see. Once they fix the bug, are we going to keep our own implemented logic or using the new one they implemented?

Comment (yogiraj07) [reply sdk/src/Handlers/AspNetCore/Internal/AWSXRayMiddleware.cs]: Once the ASP.NET Core back ports the change from NET Core 3.x to 2.x, I will upgrade the ASP.NET Core dependency and remove the custom logic. However the ETA is late June.

Comment (luluzhao) [reply sdk/src/Handlers/AspNetCore/Internal/AWSXRayMiddleware.cs]: Cool, make sure we add this into our backlog to upgrade the new dependency once they fix it.

**Pull request merged**